server/src/controller/pallotController.js

```diff
@@ -48,6 +48,33 @@ export const getPallots = async (req, res) => {
   return res.status(StatusCodes.ACCEPTED).json({ Pallot });
 };
 
+/**
+ * Gets the soonest expiring pallot
+ * @param {object} req - request for the course
+ * @param {object} res - response for the request
+ */
+export const getSoonestExpiringPallot = async (req, res) => {
+  
+  const Pallots = await prisma.pallot.findMany({ // only care about pallots that are in the warehouse
```

Comment:
rather than looping, you can just use orderBy aspect of prisma for the specific field.

https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting

Reply:
d08371cd5cb95a09676c24f0a504b56b3740ac30